File: 백준/Silver/10773. 제로/제로.cc
#define _CRT_SECURE_NO_WARNINGS
#include <stdio.h>
#include <stdlib.h>

typedef struct {
	int* arr;
	int capacity;
	int top;
} Stack;

Stack* createStack(int max) {
	Stack* S = (Stack*)malloc(sizeof(Stack));
	S->arr = (int*)malloc(sizeof(int) * max);
	S->capacity = max;
	S->top = -1;
    return S;
};

int IsEmpty(Stack* S) {
	return S->top == -1;
}

int IsFull(Stack* S) {
	return S->top == S->capacity - 1;
}

void Push(int x, Stack* S) {
	S->arr[++S->top]=x;
}

int Pop(Stack* S) {
	return S->arr[S->top--];
}

int main() {
	int num,x;
	scanf("%d", &num);
	Stack* stack = createStack(num);
	int sum = 0;
	for (int i = 0; i < num; i++) {
		scanf("%d", &x); 
		if (x != 0) {
			Push(x, stack);
		}
		else { Pop(stack); }
	}
	for (int i = 0; i <= stack->top; i++) {
		sum += stack->arr[i];
	}
	printf("%d", sum);
	free(stack->arr);
	free(stack);
	return 0;
}

File: 백준/Silver/10773. 제로/제로.java
import java.util.*;
import java.io.*;

public class Main {

	static Stack<Integer> stack = new Stack<>();
	static int sum;

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int K = Integer.parseInt(br.readLine());
		for (int i = 0; i < K; i++) {
			int k = Integer.parseInt(br.readLine());
			if (k == 0) {
				stack.pop();
			} else {
				stack.push(k);
			}
		}
		while (!stack.isEmpty()) {
			sum += stack.pop();
		}
		System.out.println(sum);
	}
}